extensions/data-plane/data-plane-google-bigquery/src/test/java/org/eclipse/edc/connector/dataplane/gcp/bigquery/pipeline/BigQueryDataSinkFactoryTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ *  Copyright (c) 2024 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.gcp.bigquery.pipeline;
+
+import org.eclipse.edc.connector.dataplane.gcp.bigquery.params.BigQueryRequestParamsProvider;
+import org.eclipse.edc.gcp.bigquery.BigQueryConfiguration;
+import org.eclipse.edc.gcp.bigquery.service.BigQueryServiceSchema;
+import org.eclipse.edc.gcp.common.GcpException;
+import org.eclipse.edc.gcp.iam.IamService;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.eclipse.edc.spi.types.domain.transfer.DataFlowStartMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.ExecutorService;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+public class BigQueryDataSinkFactoryTest {
+    private static final String TEST_PROJECT = "test-project";
+    private static final String TEST_DATASET = "test-dataset";
+    private static final String TEST_TABLE = "test-table";
+    private static final String TEST_OTHER_TYPE = "AnotherDataAddressType";
+    private static final String TEST_REQUEST_ID = "request-id";
+    private static final String TEST_PROCESS_ID = "process-id";
+    private static final String TEST_CUSTOMER_NAME = "customer-name";
+    private static final String TEST_SINK_SERVICE_ACCOUNT_NAME = "sinkAccount";
+    private BigQueryConfiguration bigQueryConfiguration = new BigQueryConfiguration(null, null, "testEndpoint", 0);
+    private TypeManager typeManager = mock();
+    private Monitor monitor = mock();
+    private ExecutorService executorService = mock();
+    private Vault vault = mock();
+    private IamService iamService = mock();
+
+    @BeforeEach
+    void setup() {
+        reset(monitor);
```

Comment:
resetting mocks is not needed here as they get instantiated anew everytime. Typically, resetting mocks is only required in very special circumstances